replica/table.cc

```diff
@@ -2096,6 +2121,10 @@ future<> tablet_storage_group_manager::handle_tablet_split_completion(const loca
 }
 
 future<> tablet_storage_group_manager::update_effective_replication_map(const locator::effective_replication_map& erm) {
+    // Exclusive lock is meant to protect storage groups, but we hold it here to prevent preemption
+    // between tablet map and storage groups updates as they need to be consistent.
+    rwlock::holder exclusive_lock = co_await get_rwlock().hold_write_lock();
+
```

Comment:
@Deexie I think it's more correct to move this excl lock to table::update_effective_replication_map() instead, as tablet map must reflect the erm, which is set in table::update_effective_replication_map(). A yield here could cause this divergence.